SignalShareExtension/ShareViewController.swift

```diff
@@ -574,9 +593,14 @@ public class ShareViewController: UIViewController, ShareViewDelegate, SAEFailed
             return Promise(error: error)
         }
 
-        // TODO Multiple attachments. In that case I'm unclear if we'll
-        // be given multiple inputItems or a single inputItem with multiple attachments.
-        guard let itemProvider: NSItemProvider = inputItem.attachments?.first as? NSItemProvider else {
+        // A single inputItem can have multiple attachments, e.g. sharing from Firefox gives
+        // one url attachment and another text attachment, where the the url would be https://some-news.com/articles/123-cat-stuck-in-tree
+        // and the text attachment would be something like "Breaking news - cat stuck in tree"
+        //
+        // FIXME: For now, we prefer the URL provider and discard the text provider, since it's more useful to share the URL than the caption
```

Comment:
With this PR we only use the URL, not the suggested text. We should include the suggested text, but it's going to be tricky to apply that change in a non-disruptive way. 

There's quite a bit of machinery based on the assumption that there's only one item provider.